api/commands/run.ts

```diff
@@ -118,7 +120,19 @@ export class Runner {
 
       // Currently, we don't support caching for operations (and any dependents)
       await this.adapter.persistStateMetadata(
-        successfulActions.filter(action => action.type !== "operation")
+        successfulActions.filter(action => action && action.type !== "operation")
+      );
+      await this.adapter.deleteStateMetadata(
```

Comment:
feels like it might be prudent to do this before we persist anything new, as it reduces the chance of the persisting failing causing the deletion to not run.

Reply:
moved it before persisting new information